src/lib/field.ts

```diff
@@ -944,20 +954,20 @@ class Field {
    * @return A {@link Field} element matching the [little endian binary representation](https://en.wikipedia.org/wiki/Endianness) of the given `bytes` array.
    */
   static fromBits(bits: (Bool | boolean)[]) {
-    let length = bits.length;
-    checkBitLength('Field.fromBits()', length);
+    const length = bits.length;
+    checkBitLength('Field.fromBits()', length, 254);
     if (bits.every((b) => typeof b === 'boolean' || b.toField().isConstant())) {
       let bits_ = bits
         .map((b) => (typeof b === 'boolean' ? b : b.toBoolean()))
         .concat(Array(Fp.sizeInBits - length).fill(false));
       return new Field(Fp.fromBits(bits_));
     }
-    let bitsVars = bits.map((b): FieldVar => {
-      if (typeof b === 'boolean') return b ? FieldVar[1] : FieldVar[0];
-      return b.toField().value;
-    });
-    let x = Snarky.field.fromBits([0, ...bitsVars]);
-    return new Field(x);
+    return bits
+      .map((b) => new Bool(b))
+      .reduce((acc, bit, idx) => {
+        const shift = 1n << BigInt(idx);
+        return acc.add(bit.toField().mul(shift));
+      }, Field.from(0));
```

Comment:
could you add `.seal()` at the end of the returned expression? that way we return a fresh variable instead of an AST of 254 summands. the latter could create an enourmous amount of unnecessary constraints when used twice in different places. see https://github.com/o1-labs/o1js/issues/1313 for more background

Reply:
Ahhh, very interesting! :)